### ISSUE TITLE
llvm-profdata should show more detailed error message.

### ISSUE THREAD

**Issue by ZequanWu** [llvm/llvm-project#57203]
If an error encountered when merging raw profile, it doesn't give more details than it should.
```
$ llvm-profdata merge -sparse base_unittests.profraw -o base_unittests.profdata
warning: base_unittests.profraw: malformed instrumentation profile data
error: no profile can be merged
```
By looking at the code, there are many places error `instrprof_error::malformed` is created with different error messages (e.g. number of value profile kinds is invalid", "function name is empty", etc), but llvm-profdata doesn't show the detailed error message.
